libazureinit/src/provision/ssh.rs

```diff
@@ -340,18 +354,13 @@ mod tests {
             },
         ];
 
-        provision_ssh(
-            &user,
-            &keys[..1],
-            Some(".ssh/xauthorized_keys".to_string()),
-        )
-        .unwrap();
-        provision_ssh(
-            &user,
-            &keys[1..],
-            Some(".ssh/xauthorized_keys".to_string()),
-        )
-        .unwrap();
+        let authorized_keys_path = user.dir.join(".ssh/xauthorized_keys");
+
+        provision_ssh(&user, &keys[1..], authorized_keys_path.clone(), false)
+            .unwrap();
```

Comment:
How come we unwrap here? Do we not want to assert anything about the state of what's returned from `provision_ssh`?

Reply:
I'm pretty sure we unwrap here to ensure provision_ssh succeeds without adding extra assertions, since the main focus of the test is verifying the content of .ssh/xauthorized_keys. Because provision_ssh should always succeed in this setup, unwrapping simplifies the test. 

Across the board, we do unwrap in all the tests because they're testing specific scenarios with the file content and unwrapping helps us quickly spot any issues with that.

With that all being said though, I don't necessarily know that we can gleam a ton of information about what happened during provision_ssh by looking only at what it returns. Given that provision_ssh only returns Ok(()) or an error, there’s limited value in adding assertions solely for its return state... I think we care more about the directories being set up correctly / whether it correctly sets permissions, writes keys, or falls back to a default path. That's what those tests are inherently checking.